GearRent/ViewControllers/DetailsViewController.m

```diff
@@ -217,14 +235,59 @@ - (void)calendar:(JTCalendarManager *)calendar didTouchDayView:(JTCalendarDayVie
     
     if([self isInDatesSelected:dayView.date]){
         [self.datesSelected removeObject:dayView.date];
-        [self setReserveButtonText];
-        [self.calendarManager reload];
-        
     } else{ // select date
         [self.datesSelected addObject:dayView.date];
-        [self setReserveButtonText];
-        [self.calendarManager reload];
     }
+    [self setReserveButtonText];
+    [self.calendarManager reload];
+}
+
+- (void)fetchDynamicPrices{
+    NSMutableArray<NSDate *> *dates = [[self.datesForDynamicPricingSet allObjects] mutableCopy];
+    self.dateRanges = [self getDynamicPriceDateRanges:dates];
+    fetchDynamicPrice(self.listing, self.dateRanges, ^(NSDictionary<NSNumber *, NSNumber *> * result, NSError * _Nonnull error) {
+        if(error == nil) {
+            NSArray *keys = result.allKeys;
+            for(NSNumber *key in keys) {
+                [self.datesToPrices setObject:[result valueForKey:key] forKey:[NSDate dateWithTimeIntervalSince1970: [key doubleValue] / 1000]];
```

Comment:
This would be more readable if you write it using modern objc syntax:

```OBJC
NSDate *date = [NSDate dateWithTimeIntervalSince1970:[key doubleValue] / 1000];
self.datesToPrices[date] = results[key]; 
```